Fix kthvalue error for scalar input

heitorschueroff opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47600 Fix kthvalue error for scalar input**

fixes https://github.com/pytorch/pytorch/issues/30818

Note that the median case was already fixed by https://github.com/pytorch/pytorch/pull/45847

Differential Revision: [D24860337](https://our.internmc.facebook.com/intern/diff/D24860337)

heitorschueroff committed:
Fix kthvalue error for scalar input

[ghstack-poisoned]

mruberry commented on test/test_torch.py:
```diff
@@ -9322,6 +9322,10 @@ def test_kthvalue(self, device, dtype):
             self.assertEqual(res1val[:, :], res2val[:, :, k - 1], atol=0, rtol=0)
             self.assertEqual(res1ind[:, :], res2ind[:, :, k - 1], atol=0, rtol=0)
 
+        # Test scalar input
```
Expand this comment with the issue it's fixing and briefly explain why the test is appropriate.

mruberry reviewed (approved):
Just one small note about improving the comment, but you also have some failing tests as a result of this change.

heitorschueroff committed:
Update on "Fix kthvalue error for scalar input"


fixes https://github.com/pytorch/pytorch/issues/30818

Note that the median case was already fixed by https://github.com/pytorch/pytorch/pull/45847

[ghstack-poisoned]

heitorschueroff committed:
Update on "Fix kthvalue error for scalar input"


fixes https://github.com/pytorch/pytorch/issues/30818

Note that the median case was already fixed by https://github.com/pytorch/pytorch/pull/45847

Differential Revision: [D24860337](https://our.internmc.facebook.com/intern/diff/D24860337)

[ghstack-poisoned]
